### PR TITLE
[fix](ub) fix uninitialized accesses in BE

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

ubsan hints:
```c++
/root/doris/be/src/olap/hll.h:93:29: runtime error: load of value 3078029312, which is not a valid value for type 'HllDataType'
/root/doris/be/src/olap/hll.h:94:23: runtime error: load of value 3078029312, which is not a valid value for type 'HllDataType'
/root/doris/be/src/runtime/descriptors.h:439:38: runtime error: load of value 118, which is not a valid value for type 'bool'
/root/doris/be/src/vec/exec/vjdbc_connector.cpp:61:50: runtime error: load of value 35, which is not a valid value for type 'bool' 
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/olap/hll.h [hunk 1]
```diff
@@ -231,12 +231,9 @@ class HyperLogLog {
 
     int64_t estimate_cardinality() const;
 
-    static std::string empty() {
+    static HyperLogLog empty() {
         static HyperLogLog hll;
```
Comment (zclllyybb) [inline be/src/olap/hll.h:235]: use RVO

File: be/src/util/bitmap_value.h [hunk 2]
```diff
@@ -1258,11 +1258,9 @@ class BitmapValue {
         return *this;
     }
 
-    static std::string empty_bitmap() {
-        std::string buf(sizeof(BitmapValue), 0);
-        BitmapValue* bitmap_value = reinterpret_cast<BitmapValue*>(buf.data());
-        bitmap_value->_type = EMPTY;
-        return buf;
+    static BitmapValue empty_bitmap() {
+        static BitmapValue bitmap_value;
```
Comment (zclllyybb) [inline be/src/util/bitmap_value.h:1262]: ditto

**Comment by zclllyybb** [comment]
test in https://github.com/apache/doris/pull/35478

**Comment by zclllyybb** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**